examples/golang/main.go

```diff
@@ -0,0 +1,292 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package config implements the functions for reading
+// configuration files and saving them into Golang objects
+
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/aws/aws-sdk-go/aws"
+
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
+)
+
+func main() {
+	var sessGetItem *session.Session
+	var sessQuery *session.Session
+
+	switch cmd := os.Args[1]; cmd {
+	case "spanner":
+		sessGetItem = createAdapterSession("GetItem")
```

Comment:
Currently the adapter requires that a session get created for each operation that needs to be performed `GetItem`, `Query`, `PutItem`, etc.

It would be better to use the `X-Amz-Target` HTTP Header field in the `/` endpoint of the adapter to determine the operation begin requested and route appropriately .  For example `X-Amz-Target: DynamoDB_20120810.Query` could be handled by `/` and routed to the query handler in the adapter.

Reply:
Captured in Issue #28

Reply:
This issue has been resolved in PR #34 and has been merged into this branch.